src/tally.F90

```diff
@@ -2417,10 +2419,29 @@ subroutine score_surface_current(p)
               t % results(1, filter_index) % value = &
                    t % results(1, filter_index) % value + p % wgt
             end if
+
+            ! IN_BOTTOM
+            if (ijk0(1) >= 1 .and. ijk0(2) >= 1 .and. ijk0(3) >= 0 .and. &
+                 ijk0(1) <= m % dimension(1) .and. &
+                 ijk0(2) <= m % dimension(2) .and. &
+                 ijk0(3) <  m % dimension(3)) then
```

Comment:
Seems like we should be checking the non-z indices before coming into the `do j` loop. You might also consider changing the logic to something like:
```Fortran
do j = ijk0(3), ijk1(3)
  if (j >= 1 .and. j <= m) then
    if (j < ijk1(3)) then
      ! score to OUT_TOP
    end if
    if (j > ijk0(3)) then
      ! score to IN_BOTTOM
    end if
  end if
end do
```